### PR TITLE
update to 3.6.1

### PR DESCRIPTION
**Pull request by tomi-font**
Just re-applied the TF-M patches and added a small MCUboot fix from https://github.com/zephyrproject-rtos/mbedtls/pull/58.

### CODE REVIEW

**Comment by ithinuel** [comment]
Do the commits like `Enable psa_can_do_hash` really need to be reverted and reapplied immediately ?

**Comment by tomi-font** [comment]
> Do the commits like `Enable psa_can_do_hash` really need to be reverted and reapplied immediately ?

Yeah, or rather this is the procedure we use. First revert the previously applied patches to be completely aligned with upstream, merge the new version to pull in, then re-apply whatever needed patches. No trying to pull in upstream commits on top of patches. Otherwise, there would be risks of (unending) merge conflicts, plus we getthe benefit that the applied patches don't get lost in the history.
Somehow Git shows the revert commits after the Mbed TLS ones (maybe because of the order they were written at, i.e. Mbed TLS ones existed already before the revert ones?), but the reverts are really done as the first thing.

**Review by valeriosetti — APPROVED** [review]
I checked that:
- up to bd3a26f9a the only difference with official Mbed TLS `v3.6.1` tag is in in the `framework` folder. This is a submodule now and it's only used for Mbed TLS testing (which is something we don't do in Zephyr), so it's OK to skip it. Patches 5b86c71a6 and 95583ee33 are for this indeed.
- all other TF-M/Zephyr patches are reverted before and re-applied after the merge of the `v3.6.1` branch of Mbed TLS.

In short IMO this is a correct update to the new 3.6.1 LTS release.

**Comment by carlescufi** [comment]
> Somehow Git shows the revert commits after the Mbed TLS ones (maybe because of the order they were written at, i.e. Mbed TLS ones existed already before the revert ones?), but the reverts are really done as the first thing.

Git always shows the commits in order they have been created, so I guess that you are probably talking about commits that come from a merge operation. Those may be displayed differently.

File: library/psa_crypto.c [hunk 1]
```diff
@@ -288,7 +288,8 @@ static uint8_t psa_get_drivers_initialized(void)
 int psa_can_do_hash(psa_algorithm_t hash_alg)
 {
     (void) hash_alg;
-    return psa_get_drivers_initialized();
+    /* Workaround for the legacy CryptoCell driver requiring hash during init */
+    return 1; //psa_get_drivers_initialized();
```
Comment (Vge0rge) [inline library/psa_crypto.c]: Its a nit but the comment after the return statement is not there in the original patch and probably added by accident, right?

**Review by Vge0rge — APPROVED** [review]
A small nit, but overall looks good

File: library/psa_crypto.c (hunk 1 above)
Comment (tomi-font) [reply library/psa_crypto.c]: It's actually there in the patch: https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/main/lib/ext/mbedcrypto/0006-Enable-psa_can_do_hash.patch

Comment (Vge0rge) [reply library/psa_crypto.c]: Oh, I was looking at the wrong revision it seems. Never mind then!

**Review by ithinuel — APPROVED** [review]
